### ISSUE TITLE
Bugsnag compatible with Java 1.7

### ISSUE THREAD

**Issue by johnrobertporter** [bugsnag/bugsnag-java#65]
I am very new to Java development and trying to integrate bugsnag with an existing program we have running that requires to be built with Java 1.7.

I notice that v3 of bugsnag for java requires Java 1.8 as I get the dreaded `Java unsupported major minor version 52.0` error, even though the release notes say it targets Java 1.6.

Can anyone help figure out how to compile bugsnag for Java 1.7?

- Which version of bugsnag should i be using?
- Are there any docs available for this version's API, being that I can only find docs for the version 3 API?

Any help would be greatly appreciated.

**Comment by johnrobertporter** [comment]
Ok, so we got version `3.0.0` to compile, but not the latest `3.1.1`. Can anyone help me to understand what the differences are?

I need to learn more about how these APIs work.

Thanks in advance.

**Comment by fractalwrench** [comment]
Thanks for the report @johnrobertporter - which of the installation methods are you using to integrate Bugsnag into your application? A full list of the options is available [here](https://docs.bugsnag.com/platforms/java/other/#installation).

**Comment by johnrobertporter** [comment]
Hi @fractalwrench, we're using Maven to build the war file and it says version v3.1 for the compiler plugin, and v2.3 for the war plugin.

The JDK we got working was OpenJDK v1.7

Initially we had this in the pom file:

```xml
<dependency>
  <groupId>com.bugsnag</groupId>
  <version>3.1.1</version>
  <artifactId>bugsnag</artifactId>
</dependency>
```

but had to switch it to:

```xml
<dependency>
  <groupId>com.bugsnag</groupId>
  <version>3.0</version>
  <artifactId>bugsnag</artifactId>
</dependency>
```

which compiled fine.

Like I say, we're not Java people here and kind of learning as we go, so if there's anything else I can give in the way of info, please ask.

Thanks for looking at this :)

**Comment by fractalwrench** [comment]
@johnrobertporter we've just released v3.1.2, which should resolve some packaging issues with the Maven Artefact, and restore JDK 7 support. Sorry about this - our CI server stopped supporting Java 7 so we missed this.

Please get in touch if you have any other questions about getting setup and I'll look into it as a priority.

**Comment by johnrobertporter** [comment]
Thats fantastic news!! Thanks for your help, we'll get it upgraded asap and report back if there are any issues.
